pkg/pod/entrypoint.go

```diff
@@ -120,6 +120,7 @@ func orderContainers(entrypointImage string, extraEntrypointArgs []string, steps
 		}
 		argsForEntrypoint = append(argsForEntrypoint, extraEntrypointArgs...)
 		argsForEntrypoint = append(argsForEntrypoint, resultArgument(steps, results)...)
+		argsForEntrypoint = append(argsForEntrypoint, "-timeout", stepTimeouts[i])
```

Comment:
It might be easier to only append the arg if it's value is !=0 -- in particular, that would remove the need for a lot of (all of?) the changes in `entrypoint_test.go` below.